### PR TITLE
Handle WooCommerce specific queries to display results through ES

### PR DESCRIPTION
**Pull request by joshuaabenazer**
- Include child terms to the Elasticsearch query when querying a parent term of a hierarchical taxonomy
- Fix issue of product ordering
- Fixes https://github.com/10up/elasticpress-woocommerce/issues/3
- Handle WC widget queries.

### CODE REVIEW

File: elasticpress-woocommerce.php [hunk 1]
```diff
@@ -265,10 +280,8 @@ function epwc_translate_args( $query ) {
 			$query->set( 'meta_query', $meta_query );
 		}
 
-		$post_parent = $query->get( 'post_parent', false );
-
-
-		if ( 'product' === $post_type ) {
+		// Assuming empty to be product post type
+		if ( empty( $post_type ) ) {
```
Comment (tlovett1) [inline elasticpress-woocommerce.php]: Can you explain this part?

Comment (joshuaabenazer) [inline elasticpress-woocommerce.php]: I have changed that line to include the product post type checking too. The check for empty is added as this is empty on the product cat archives and we do not want the meta query to execute on all the remaining WooCommerce post type queries.

File: elasticpress-woocommerce.php [hunk 2]
```diff
@@ -296,13 +309,19 @@ function epwc_translate_args( $query ) {
...
 						break;
 					default:
-						$query->set( 'orderby', 'menu_order title date meta._price.long' ); // Order by menu and title.
+						$query->set( 'orderby', 'menu_order title date' ); // Order by menu and title.
 						$query->set( 'order', 'asc' );
 				}
 			} else {
-				$query->set( 'orderby', 'menu_order title date meta._price.long' ); // Order by menu and title.
-				$query->set( 'order', 'asc' );
+					$query->set( 'orderby', 'menu_order title date' ); // Order by menu and title.
```
Comment (tlovett1) [inline elasticpress-woocommerce.php]: Line tabbing is off

File: elasticpress-woocommerce.php [hunk 3]
```diff
@@ -296,13 +309,19 @@ function epwc_translate_args( $query ) {
...
+					$query->set( 'order', 'asc' );
 			}
+		} // Conditional check for orders
+		elseif ( in_array( $post_type, array( 'shop_order', 'shop_order_refund' ) ) || ( is_array( $post_type ) && ! array_diff( $post_type, $supported_post_types ) ) ) {
```
Comment (tlovett1) [inline elasticpress-woocommerce.php]: The second part of the conditional is confusing. Perhaps it should be it's own conditional?

**Pull request merged**